doc/lua_api.txt

```diff
@@ -234,6 +235,40 @@ stripping out the file extension:
 * e.g. `foomod_foothing.png`
 * e.g. `foomod_foothing`
 
+Texture pack directory structure
```

Comment:
lua_api.txt doesn't seem the right place for this considering there are other files in the doc/ dir as well. Maybe create a doc/texture_packs.txt file?